lib/internal/webstreams/adapters.js

```diff
@@ -0,0 +1,832 @@
+        return closed.promise;
+      }
+
+      controller = undefined;
+      return PromiseResolve();
+    },
+  }, strategy);
+}
+
+/**
+ * @param {WritableStream} writableStream
+ * @param {{
+ *   decodeStrings? : boolean,
+ *   highWaterMark? : number,
+ *   objectMode? : boolean,
+ *   signal? : AbortSignal,
+ * }} [options]
+ * @returns {Writable}
+ */
+function newStreamWritableFromWritableStream(writableStream, options = {}) {
+  if (!isWritableStream(writableStream)) {
+    throw new ERR_INVALID_ARG_TYPE(
+      'writableStream',
+      'WritableStream',
+      writableStream);
+  }
+
+  validateObject(options, 'options');
+  const {
+    highWaterMark,
+    decodeStrings = true,
+    objectMode = false,
+    signal,
+  } = options;
+
+  validateBoolean(objectMode, 'options.objectMode');
+  validateBoolean(decodeStrings, 'options.decodeStrings');
+  if (signal !== undefined)
+    validateAbortSignal(signal, 'options.signal');
+
+  const writer = writableStream.getWriter();
+  let closed = false;
+
+  const writable = new Writable({
+    highWaterMark,
+    objectMode,
+    decodeStrings,
+    signal,
+
+    writev(chunks, callback) {
+      function done(error) {
+        try {
+          callback(error);
+        } catch (error) {
+          process.nextTick(() => writable.destroy(error));
+        }
+      }
+
+      PromisePrototypeThen(
+        writer.ready,
+        () => {
+          return PromisePrototypeThen(
+            PromiseAll(
+              ArrayPrototypeMap(
+                chunks,
+                (chunk) => writer.write(chunk))),
+            done,
+            done);
+        },
+        done);
+    },
+
+    write(chunk, encoding, callback) {
+      if (typeof chunk === 'string' && decodeStrings && !objectMode)
+        chunk = new Uint8Array(Buffer.from(chunk, encoding));
+
+      function done(error) {
+        try {
+          callback(error);
+        } catch (error) {
+          writable.destroy(error);
+        }
+      }
+
+      PromisePrototypeThen(
+        writer.ready,
+        () => {
+          return PromisePrototypeThen(
+            writer.write(chunk),
+            done,
+            done);
+        },
+        done);
+    },
+
+    destroy(error, callback) {
+      function done() {
+        try {
+          callback(error);
+        } catch (error) {
+          process.nextTick(() => { throw error; });
+        }
+      }
+
+      if (!closed) {
+        if (error != null) {
+          PromisePrototypeThen(
+            writer.abort(error),
+            done,
+            done);
+        } else {
+          PromisePrototypeThen(
+            writer.close(),
+            done,
+            done);
+        }
+        return;
+      }
+
+      done();
+    },
+
+    final(callback) {
+      function done(error) {
+        try {
+          callback(error);
+        } catch (error) {
+          process.nextTick(() => writable.destroy(error));
+        }
+      }
+
+      if (!closed) {
```

Comment:
```suggestion
      if (!stream.isFinished(writable) {
```

https://github.com/nodejs/node/pull/39294

Reply:
`closed` here is tracking the closed state of the web stream, not the node.js stream.